pkg/fakerp/network.go

```
@@ -0,0 +1,85 @@
+	managementResourceGroup string
+	log                     *logrus.Entry
+}
+
+func newNetworkManager(ctx context.Context, log *logrus.Entry, subscriptionID, resourceGroupName string) (*networkManager, error) {
+	authorizer, err := azureclient.GetAuthorizerFromContext(ctx, api.ContextKeyClientAuthorizer)
+	if err != nil {
+		return nil, err
+	}
+
+	return &networkManager{
+		pec:                     network.NewPrivateEndpointsClient(ctx, log, subscriptionID, authorizer),
+		nic:                     network.NewInterfacesClient(ctx, log, subscriptionID, authorizer),
+		plsc:                    network.NewPrivateLinkServicesClient(ctx, log, subscriptionID, authorizer),
+		managementResourceGroup: resourceGroupName,
+		log:                     log,
+	}, nil
+}
+
+func (nm *networkManager) deletePLSPE(ctx context.Context, resourceGroupname, resourceName string) error {
+	pls, err := nm.plsc.Get(ctx, resourceGroupname, resourceName, "")
+	if err != nil {
+		return err
+	}
+	for _, pe := range *pls.PrivateEndpointConnections {
+		resources, err := azure.ParseResourceID(*pe.PrivateEndpointConnectionProperties.PrivateEndpoint.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = nm.pec.Delete(ctx, nm.managementResourceGroup, resources.ResourceName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func (nm *networkManager) getPrivateEndpointIP(ctx context.Context, resourceName string) (*string, error) {
+	pe, err := nm.pec.Get(ctx, nm.managementResourceGroup, resourceName, "")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, nic := range *pe.NetworkInterfaces {
+		resource, err := azure.ParseResourceID(*nic.ID)
+		if err != nil {
+			return nil, err
+		}
+		ni, err := nm.nic.Get(ctx, resource.ResourceGroup, resource.ResourceName, "")
+		if err != nil {
+			return nil, err
+		}
+		for _, ip := range *ni.IPConfigurations {
+			if ip.PrivateIPAddress != nil {
+				return ip.PrivateIPAddress, nil
+			}
+		}
+	}
+
+	return nil, err
```

should return a synthetic error here